src/markd/parsers/inline.cr

```diff
@@ -746,13 +746,15 @@ module Markd::Parser
     end
 
     def normalize_uri(uri : String)
-      URI.escape(decode_uri(uri)) do |byte|
+      io = String::Builder.new
```

Comment:
It's usually better to use `String.build do |io|` instead of instantiating `String::Builder` directly.

Reply:
The string builder issue has been fixed.

Reply:
Without the the custom encoding logic the following tests fail, so the custom encoding maybe could be modified, but not eliminated...

```
1) Links - 38
[foo<http://example.com/?search=](uri)>

       Expected: "<p>[foo<a href=\"http://example.com/?search=%5D(uri)\">http://example.com/?search=](uri)</a></p>\n"
            got: "<p>[foo<a href=\"http://example.com/?search=](uri)\">http://example.com/?search=](uri)</a></p>\n"

     # fixtures/spec.txt:7546

2) Links - 50
[foo<http://example.com/?search=][ref]>

[ref]:␣/uri

       Expected: "<p>[foo<a href=\"http://example.com/?search=%5D%5Bref%5D\">http://example.com/?search=][ref]</a></p>\n"
            got: "<p>[foo<a href=\"http://example.com/?search=][ref]\">http://example.com/?search=][ref]</a></p>\n"

     # fixtures/spec.txt:7699

3) Autolinks - 9
<http://example.com/\[\>

       Expected: "<p><a href=\"http://example.com/%5C%5B%5C\">http://example.com/\\[\\</a></p>\n"
            got: "<p><a href=\"http://example.com/%5C[%5C\">http://example.com/\\[\\</a></p>\n"

     # fixtures/spec.txt:8445
```